sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -154,6 +154,26 @@ private[sql] trait LookupCatalog extends Logging {
     }
   }
 
+  object AsFunctionIdentifier {
+    def unapply(parts: Seq[String]): Option[FunctionIdentifier] = {
+      def namesToFunctionIdentifier(names: Seq[String]): Option[FunctionIdentifier] = names match {
+        case Seq(name) => Some(FunctionIdentifier(name))
+        case Seq(database, name) => Some(FunctionIdentifier(name, Some(database)))
+        case _ => None
+      }
+      parts match {
+        case CatalogAndMultipartIdentifier(None, names)
+          if CatalogV2Util.isSessionCatalog(currentCatalog) =>
+          namesToFunctionIdentifier(names)
+        case CatalogAndMultipartIdentifier(Some(catalog), names)
+          if CatalogV2Util.isSessionCatalog(catalog) &&
+            CatalogV2Util.isSessionCatalog(currentCatalog) =>
```

Comment:
Why is the current catalog also checked? Can't I refer to `spark_catalog.db.func` and get a `FunctionIdentifier` back?

Reply:
I followed `AsTableIdentifier` for this check and see related comments [here](https://github.com/apache/spark/pull/30919#discussion_r549610277). But now looking at it, I think it should be OK to remove this check for both function and table identifier.